### ISSUE TITLE
GitBucket Compatibility Issues With golang

### ISSUE THREAD

**Issue by tundratim** [gitbucket/gitbucket#1515]
I have a client whose dev team has identified what is believed to be a golang import problem that makes it not place nice with GitBucket.  See:

https://github.com/golang/go/issues/19766

Additionally, there is a corner case that has shown up that they believe is an artifact of default GitBucket behavior:

If you try to import domain.of.repo/repo/name/deep/import, go tries to http get domain.of.repo/repo/name/deep/import, which 404s, because gitbucket serves that path at domain.of.repo/repo/name/tree/master/deep/import.

The devs are asking if the GitBucket team might consider whether gitbucket could be enhanced to be smart enough to redirect requests for a path inside a known repo to default to the master branch
so http get /repo/name/deep/import -> /repo/name/tree/master/deep/import

Thoughts?

**Comment by McFoggy** [comment]
@tundratim I think a plugin would be able to do this probably by installing a servlet filter detecting `?go-get=1` and making the redirection.

**Comment by tundratim** [comment]
@McFoggy Yes, I don't want to see GB default behavior diverge from what it is today.  I don't know scala (or java very well for that matter).  Could you suggest a place to get started writing a plugin/servlet?

Thanks!

**Comment by aadrian** [comment]
>  Could you suggest a place to get started writing a plugin/servlet?

1. Here https://gitter.im/gitbucket/gitbucket group chat if you have a specific question.

2. The first article about developing GitBucket plug-ins http://gitbucket.github.io/gitbucket-news/gitbucket/2015/06/29/how-to-create-plugin.html (it's old but still good)
3. A more actual introduction https://gitbucket.github.io/gitbucket-news/gitbucket/2017/01/10/lets-begin-scala-with-gitbucket.html  (I would skip the JRebel part since it's more cumbersome to get it work)
4. Take a look at the existing plug-ins http://gitbucket-plugins.github.io/  - **all** have sources so they're  a good starting point.

> I don't know scala (or java very well for that matter)

If you use Linux/OSX it's easier by using this: http://sdkman.io/  to get Java, Scala and SBT in order to be able to quickly and correctly get setup with everything.
